packages/customer-erasure/src/main.js

```diff
@@ -0,0 +1,217 @@
+        const reference = CustomerErasure.buildReference(ids)
+        const messagesUri = requestBuilder.messages.where(reference).build()
+        const request = CustomerErasure.buildRequest(messagesUri, 'GET')
+
+        const messages = await this._getAllMessages(request)
+
+        results = [...messages, ...results]
+      }
+      this.logger.info('Export operation completed successfully')
+      return Promise.resolve(results)
+    })
+  }
+
+  async _getAllMessages(request) {
+    const messages = await this.client.process(request, payload => {
+      if (payload.statusCode !== 200 && payload.statusCode !== 404)
+        return Promise.reject(
+          Error(`Request returned status code ${payload.statusCode}`)
+        )
+
+      return Promise.resolve(payload)
+    })
+    return flatten(messages.map(result => result.body.results))
+  }
+
+  deleteAll(customerId) {
+    if (!customerId) throw Error('missing `customerId` argument')
+    this.logger.info('Starting deletion')
+
+    const requestBuilder = createRequestBuilder({
+      projectKey: this.apiConfig.projectKey,
+    })
+    const customersUri = {
+      uri: requestBuilder.customers.where(`id = "${customerId}"`).build(),
+      builder: requestBuilder.customers,
+    }
+    const ordersUri = {
+      uri: requestBuilder.orders.where(`customerId = "${customerId}"`).build(),
+      builder: requestBuilder.orders,
+    }
+    const cartsUri = {
+      uri: requestBuilder.carts.where(`customerId = "${customerId}"`).build(),
+      builder: requestBuilder.carts,
+    }
+    const paymentsUri = {
+      uri: requestBuilder.payments
+        .where(`customer(id = "${customerId}")`)
+        .build(),
+      builder: requestBuilder.payments,
+    }
+    const shoppingsListsUri = {
+      uri: requestBuilder.shoppingLists
+        .where(`customer(id = "${customerId}")`)
+        .build(),
+      builder: requestBuilder.shoppingLists,
+    }
+    const reviewsUri = {
+      uri: requestBuilder.reviews
+        .where(`customer(id = "${customerId}")`)
+        .build(),
+      builder: requestBuilder.reviews,
+    }
+
+    const urisOfResourcesToDelete = [
+      customersUri,
+      ordersUri,
+      cartsUri,
+      paymentsUri,
+      shoppingsListsUri,
+      reviewsUri,
+    ]
+
+    return Promise.all(
+      urisOfResourcesToDelete.map(uri => {
+        const request = CustomerErasure.buildRequest(uri.uri, 'GET')
+
+        return this.client.process(request, payload => {
+          if (payload.statusCode !== 200 && payload.statusCode !== 404)
+            return Promise.reject(
+              Error(`Request returned status code ${payload.statusCode}`)
+            )
+          if (payload.statusCode === 200) this._deleteOne(payload, uri.builder)
+          return Promise.resolve()
+        })
+      })
+    )
+  }
+
+  _deleteOne(payload, builder) {
+    const results = payload.body.results
+    if (results.length > 0) {
+      results.forEach(async result => {
```

Comment:
The `this.client.execute(deleteRequest)` is a promise, so that part should be awaited.

Actually it should probably be a `Promise.all(results.map(...))`